components/ui/JsonLd.tsx

```
@@ -0,0 +1,38 @@
+    // Escape the opening bracket to avoid closing the <script> tag when
+    // the JSON contains '</script>' sequences.
+    .replace(/</g, "\\u003c")
+    // Escape line separator characters which are valid in JS strings but
+    // break inline script embedding in HTML.
+    .replace(/\u2028/g, "\\u2028")
+    .replace(/\u2029/g, "\\u2029")
+    // Optionally escape HTML comment closers to be extra-safe.
+    .replace(/-->/g, "--\\u003e");
+
+  return (
+    <script type="application/ld+json" dangerouslySetInnerHTML={{ __html: json }} />
```

Biome (2.1.2)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)